flocker/node/_loop.py

```diff
@@ -275,14 +275,15 @@ def output_STORE_INFO(self, context):
             context.client, context.configuration, context.state)
 
     def output_CONVERGE(self, context):
-        d = DeferredContext(self.deployer.discover_local_state())
+        # XXX known_local_state = lookup NodeState for this node. If it
+        # doesn't exist, create NodeState with all values set to None
+        # (i.e. we know nothing).
+        d = DeferredContext(self.deployer.discover_local_state(
+            known_local_state))
 
         def got_local_state(local_state):
             # Current cluster state is likely out of date as regards the local
             # state, so update it accordingly.
-            # XXX FLOC-1542 will need to deal with partial updates, and
-            # make sure there is no duplication with code in
-            # flocker.control._clusterstate.
```

Comment:
Something more necessary here?  Call `cluster_state.update_node` instead of `cluster_state.set`?

Pre-existing issue - the attribute is named `cluster_state` but the documentation for the class says `state` and that it is a `Deployment` but I think it's a `DeploymentState` now.